### PR TITLE
cam/crank sync improvements

### PR DESCRIPTION
**Pull request by mck1117**
- ~~Clarify `Engine::getOperationMode` and `doesTriggerImplyOperationMode`~~ now #4153
- Don't switch from wasted -> sequential until we get VVT phase information
- ~~Set a maximum RPM above which we don't try to use the cam to help sync the crank~~ now #4156 
- ~~Encapsulate `vvtSyncCounter`~~ now #4154

related #4099

### CODE REVIEW

**Comment by rusefillc** [comment]
Please rebase since https://github.com/rusefi/rusefi/commit/32d46d1f097d29160d4fa5047186004fae88a1e3

Please CHANGELOG

File: firmware/util/timer.cpp [hunk 1]
```diff
@@ -2,18 +2,10 @@
...
-Timer::Timer() {
-	init();
-}
-
 void Timer::reset() {
 	m_lastReset = getTimeNowNt();
 }
 
-void Timer::init() {
-	m_lastReset = INT64_MIN / 8;
```
Comment (rusefillc) [inline firmware/util/timer.cpp:14]: i believe that this API is needed regardless, at the moment we cannot get timer back into original state

**Comment by rusefillc** [comment]
how many atomic changes are here total? :)

**Comment by mck1117** [comment]
> how many atomic changes are here total? :)

hehe, a few :)

**Comment by rusefillc** [comment]
Thank you for four pieces of progress!!!

File: unit_tests/tests/trigger/test_trigger_decoder_2.cpp [hunk 2]
```diff
@@ -225,3 +229,76 @@ TEST(TriggerDecoder, NotEnoughTeeth_CausesError) {
 	EXPECT_FALSE(dut.getShaftSynchronized());
 	EXPECT_EQ(0, dut.currentCycle.current_index);
 }
+
+TEST(TriggerDecoder, PrimaryDecoderNoDisambiguation) {
```
Comment (mck1117) [inline unit_tests/tests/trigger/test_trigger_decoder_2.cpp:233]: there we go, finally had enough other stuff in place to write these tests!

File: firmware/controllers/math/engine_math.cpp [hunk 3]
```diff
@@ -410,9 +410,15 @@ void prepareIgnitionPinIndices(ignition_mode_e ignitionMode) {
 ignition_mode_e getCurrentIgnitionMode() {
 	ignition_mode_e ignitionMode = engineConfiguration->ignitionMode;
 #if EFI_SHAFT_POSITION_INPUT
-	// In spin-up cranking mode we don't have full phase sync. info yet, so wasted spark mode is better
-	if (ignitionMode == IM_INDIVIDUAL_COILS && engine->rpmCalculator.isSpinningUp())
-		ignitionMode = IM_WASTED_SPARK;
+	// In spin-up cranking mode we don't have full phase sync info yet, so wasted spark mode is better
+	if (ignitionMode == IM_INDIVIDUAL_COILS) {
+		bool missingPhaseInfoForSequential = 
+			!engine->triggerCentral.triggerState.hasSynchronizedPhase();
```
[the remark this answers is not in the document]
Comment (mck1117) [reply firmware/controllers/math/engine_math.cpp:416]: but it does now!

**Pull request merged**